### PR TITLE
feat(datasource/custom): allow to fetch from YAML data endpoint

### LINKED ISSUE #25634: feat(datasource/custom): allow to fetch from YAML data endpoint
### Describe the proposed change(s).

Similar to the current `json` implementation this new `yaml` option should query an http endpoint tough instead a yaml parser should be used

Notes: 
- the request should send the `Accept` header with `application/yaml` as value
- After parsing as yaml, jsonata rules should be applied

### PR DESCRIPTION
**Pull request by nSimonFR**
## Context

Closes #25634
 
## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by maximeancellin — APPROVED** [review]

**Review by maximeancellin — APPROVED** [review]
LGTM ! ✅

**Review by maximeancellin — APPROVED** [review]

**Review by maximeancellin — APPROVED** [review]

File: lib/config/types.ts [hunk 1]
```diff
@@ -513,3 +513,4 @@ export interface ValidationResult {
   errors: ValidationMessage[];
   warnings: ValidationMessage[];
 }
+
```
Comment (viceice) [inline lib/config/types.ts:516]: please avoid such changes

File: lib/modules/datasource/custom/formats/yaml.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { newlineRegex } from '../../../../util/regex';
+import type { ReleaseResult } from '../../types';
+
+export async function fetch(
+  http: Http,
+  url: string,
+): Promise<ReleaseResult | null> {
+  const response = await http.get(url);
+
+  return yaml.load(response.body.replace(newlineRegex, '\n')) as ReleaseResult;
```
Comment (viceice) [inline lib/modules/datasource/custom/formats/yaml.ts:13]: why replace the newline? the yaml parser should handle windows newlines.

File: lib/modules/datasource/custom/index.spec.ts [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import { describe, it } from '@jest/globals';
+import expect from 'expect';
```
Comment (nSimonFR) [inline lib/modules/datasource/custom/index.spec.ts:2]: Self-reviewing: unwanted ? My vscode auto imported them

**Review by secustor — CHANGES_REQUESTED** [review]
Thanks for contributing!

**Comment by secustor** [comment]
Also needs [prettier-fix](https://github.com/renovatebot/renovate/blob/main/package.json#L45)

File: lib/modules/datasource/custom/index.spec.ts [hunk 4]
```diff
@@ -198,6 +198,41 @@ describe('modules/datasource/custom/index', () => {
...
+          {
+            version: '2.0.0',
+          },
+          {
+            version: '3.0.0',
+          },
+        ],
+      };
+
+      const yaml = `releases:\n  - version: 1.0.0\n  - version: 2.0.0\n  - version: 3.0.0`;
```
Comment (secustor) [inline lib/modules/datasource/custom/index.spec.ts:216]: Use a codeblock to make this more readable https://github.com/renovatebot/renovate/blob/2ce4fc5b06055a08108fbb5c5437f522d8371097/lib/util/exec/hermit.spec.ts#L65

File: lib/modules/datasource/custom/readme.md [hunk 5]
```diff
@@ -118,6 +118,39 @@ When Renovate receives this response with the `plain` format, it will convert it
 
 After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied as usual to further process the JSON data.
 
+### Yaml
+
+If `yaml` is used, response is processed by the `ts-yaml` library and then is converted into JSON.
```
Comment (secustor) [inline lib/modules/datasource/custom/readme.md:123]:
```suggestion
If `yaml` is used, response is parsed and converted into JSON for further processing.
```

**Review by secustor — CHANGES_REQUESTED** [review]

**Review by secustor — APPROVED** [review]

**Review by viceice — APPROVED** [review]

File: lib/modules/datasource/custom/readme.md [hunk 6]
```diff
@@ -118,6 +118,39 @@ When Renovate receives this response with the `plain` format, it will convert it
 
 After the conversion, any `jsonata` rules defined in the `transformTemplates` section will be applied as usual to further process the JSON data.
 
+### Yaml
```
Comment (astellingwerf) [inline lib/modules/datasource/custom/readme.md:121]: This should be at h4 instead of h3 to be on the same as JSON and plain

**Pull request merged**